app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/MainActivity.kt

```diff
@@ -1,23 +1,37 @@
 package com.puzzlebench.clean_marvel_kotlin.presentation
 
 import android.os.Bundle
+import android.view.View
 import com.puzzlebench.clean_marvel_kotlin.R
 import com.puzzlebench.clean_marvel_kotlin.data.service.CharacterServicesImpl
 import com.puzzlebench.clean_marvel_kotlin.domain.usecase.GetCharacterServiceUseCase
 import com.puzzlebench.clean_marvel_kotlin.presentation.base.BaseRxActivity
+import com.puzzlebench.clean_marvel_kotlin.presentation.mvp.CharacterModel
 import com.puzzlebench.clean_marvel_kotlin.presentation.mvp.CharacterPresenter
 import com.puzzlebench.clean_marvel_kotlin.presentation.mvp.CharacterView
 import io.realm.Realm
+import kotlinx.android.synthetic.main.activity_main.*
 
 open class MainActivity : BaseRxActivity() {
 
     val getCharacterServiceUseCase = GetCharacterServiceUseCase(CharacterServicesImpl())
-    val presenter = CharacterPresenter(CharacterView(this), getCharacterServiceUseCase, subscriptions)
+    val presenter = CharacterPresenter(CharacterView(this), CharacterModel(), getCharacterServiceUseCase, subscriptions)
```

Comment:
delete the subscriptions.. and put the useCase as a parameter in the model

Reply:
Done